pymc3/distributions/discrete.py

```diff
@@ -710,19 +710,9 @@ def __init__(self, p, *args, **kwargs):
         self.p = (p.T / tt.sum(p, -1)).T
         self.mode = tt.argmax(p)
 
-    def _random(self, k, p, size=None):
-        if len(p.shape) > 1:
-            return np.asarray(
-                [np.random.choice(k, p=pp, size=size)
-                    for pp in p]
-            )
-        else:
-            return np.asarray(np.random.choice(k, p=p, size=size))
-
     def random(self, point=None, size=None):
         p, k = draw_values([self.p, self.k], point=point, size=size)
```

Comment:
For my own understanding why is draw_values called here? Why can't we just use the generate samples function directly?

Reply:
I realize this may be a really basic question, but I'm biasing towards asking rather than assuming as I learn more about pymc3 :)

Reply:
sorry, i think this was a question for me: `draw_values` is generally to get the parameters of a distribution to the right shape (`size=size`), with the right conditioning (`point=point`).  The `generate_samples` function actually draws new samples for an RV. 

If you trace the code, `draw_values` will call `.random` on each of the variables, which will (usually) end up at `generate_samples`.  This is how we traverse back up the computation graph.